### ISSUE TITLE
FYI: installing on MacOS Ventura (13.0.1) fails on grpcio

### ISSUE THREAD

**Issue by georgettica** [pulumi/pulumi#11424]
### What happened?

followed the instructions and failed on grpcio

### Steps to reproduce

pulumi new aws-python

### Expected Behavior

that is will create the stack

### Actual Behavior

tried to pull the full log but it seems I cannot sanitize it correctly, so I will work on that tomorrow, sry about that

### Output of `pulumi about`

```
$ pulumi about
CLI          
Version      3.47.1
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME    VERSION
python  unknown

Host     
OS       darwin
Version  13.0.1
Arch     arm64

This project is written in python: executable='/opt/homebrew/bin/python3' version='3.10.8
'

Backend        
Name           YYY
URL            file://
User           XXX
Organizations  

Dependencies:
NAME        VERSION
dill        0.3.6
parver      0.4.0
pip         22.3.1
protobuf    4.21.9
PyYAML      6.0.0
semver      2.13.0
setuptools  65.6.0
six         1.16.0
wheel       0.38.4

Pulumi locates its logs in XXYYZZ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Guessing the repro here might be just "pulumi new was-python && pulumi up" on an M1-powered (arm) Mac machine.

**Comment by justinvp** [comment]
Just want to confirm that `pulumi new aws-python && pulumi up` works OK on an M1 mac on Monterey. Checking to see if anyone on the team has already upgraded to Ventura, otherwise, we can look into running in a VM (or possibly CI) to repro.

**Comment by RobbieMcKinstry** [comment]
Also confirmed that `pulumi new aws-python && pulumi up` works on my M1 mac on `Ventura 13.0.1`.

https://app.warp.dev/block/NXK69cORCxjbQBVp9IYcrZ

**Comment by RobbieMcKinstry** [comment]
I compiled Pulumi from source.

**Comment by t0yv0** [comment]
Apologies, reopening. I misunderstood this as latest version of Pulumi.

**Comment by joeduffy** [comment]
marking p1 per discussion with @lukehoban as this is blocking usage on a common configuration.

**Comment by kpitzen** [comment]
@joeduffy @georgettica @t0yv0 @RobbieMcKinstry @justinvp  I believe this should be fixed in our next release.  #11431 was merged, and I was able to both repro the original issue pre-fix, and verify its resolution post-fix in Python 3.11 on an M1 machine.

**Comment by RobbieMcKinstry** [comment]
Since we believe this fixed, I'll close the issue again and apply the label. Feel free to reopen if Pulumi v3.50.2 does not solve the problem.
